10_binary_counter.md

```diff
@@ -319,12 +317,10 @@ It's a carry propagation.
     1 2  ...   32           1 2  ...   32
     y 0  ...   0    -->     0 y  ...   0
```

Comment:
Something I'm still unsure about is this: The text above says

> The element in the `nth` bit will be the one that has had `n` victories.

But that's not actually consistent with the behaviour described in these examples.

When an element first enters the counter at the left-hand end, it has had _no_ victories. Which implies that the first element in the counter should be labelled '0', rather than '1'.

Then we see the example of a new element 'x' being introduced to the counter, when 'y' is already in the first slot. If x>y then it _replaces_ 'y'. So it goes in to a slot that is normally occupied by elements that have had _no_ victories, even though it has actually had one victory.

Conversely if y>x then 'y' _does_ bump up to the next slot.

My point here is that the state of the counter is different depending on the _order_ in which you introduce the elements 'x' and 'y'. Which doesn't seem to match the initial explanation of what the 'power' of an element means.

An explanation of that would be welcome!

EDIT: OK, I see now that the `add_to_counter` implementation *always* promotes the winner in to the next slot, which makes sense. So I think the examples with 'x' and 'y' should probably be updated to reflect that. And probably the first column should be labelled '0' instead of '1' to show that elements in that slot have "played no games" yet.

Reply:
It should be the behavior you described in the EDIT. I will ensure the examples are not contrary and shift to 0 based indexing.